### PR TITLE
Update release instructions to ease onboarding

### PR DESCRIPTION
**Pull request by willb**
This commit:

1.  Adds instructions for authenticating to GCP,
2.  adds instructions for setting the right kubectl context, and
3.  ensures that workflow names are less than 63 characters.

Related to
#215 Minimal release process
#506 Cut a 0.1 release

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/493)

### CODE REVIEW

**Comment by willb** [comment]
I signed it!

(I hadn't added my "official" corporate email to my GitHub account.)

**Comment by willb** [comment]
@jlewi PTAL; I think we can remove the WIP from this.

**Comment by willb** [comment]
(After this is merged, we can branch a `v0.1-branch` and update the instructions to refer to specific SHAs there.)

File: releasing.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
 
 Some preliminary instructions for how to cut a release.
 
+## Authenticate to GCP
+
+If you're new to using GKE or are new to the release team, you'll need to authenticate to GCP first.  [Install the `gcloud` tool](https://cloud.google.com/sdk/gcloud/) and then execute the following commands, substituting your release team account for `your-team-account@kubeflow.org`:
+
+```
+gcloud config set account your-team-account@kubeflow.org
```
Comment (jlewi) [inline releasing.md]: gcloud config set account <your google account>

Most people won't have kubeflow.org accounts and this makes it seem like they need one.
They don't.

They just need to be added to 
release-team@kubeflow.org

File: releasing.md [hunk 2]
```diff
@@ -104,13 +127,14 @@ to find the latest green postsubmit.
...
 git checkout ${COMMIT}
 cd components/tensorflow-notebook-image/releaser
 ```
 
 ```
-PULL_BASE_SHA=<commit to build>
+PULL_BASE_SHA=${COMMIT:0:8}
 DATE=`date +%Y%m%d`
 VERSION_TAG="v${DATE}-${PULL_BASE_SHA}"
 JOB_NAME="tensorflow-notebook-image-release"
```
Comment (jlewi) [inline releasing.md]: Can you add instructions for cutting a branch?

**Pull request merged**